autogpt_platform/frontend/src/components/auth/WaitlistErrorContent.tsx

```diff
@@ -0,0 +1,59 @@
+import { Button } from "../atoms/Button/Button";
+import { Text } from "../atoms/Text/Text";
+
+interface WaitlistErrorContentProps {
+  onClose: () => void;
+  closeButtonText?: string;
+  closeButtonVariant?: "primary" | "secondary";
+}
+
+export function WaitlistErrorContent({
+  onClose,
+  closeButtonText = "Close",
+  closeButtonVariant = "primary",
+}: WaitlistErrorContentProps) {
+  return (
+    <div className="flex flex-col items-center gap-6">
+      <Text variant="h3">Join the Waitlist</Text>
+      <div className="flex flex-col gap-4 text-center">
+        <Text variant="large-medium" className="text-center">
+          The AutoGPT Platform is currently in closed beta. Your email address
+          isn&apos;t on our current allowlist for early access.
+        </Text>
+        <Text variant="body" className="text-center">
+          Join our waitlist to get notified when we open up access!
+        </Text>
+      </div>
+      <div className="flex gap-3">
+        <Button
+          variant="secondary"
+          onClick={() => {
+            window.open("https://agpt.co/waitlist", "_blank");
+          }}
+        >
+          Join Waitlist
+        </Button>
+        <Button variant={closeButtonVariant} onClick={onClose}>
+          {closeButtonText}
+        </Button>
```

Comment:
nit: make the params for this optional so we can omit it if it just says "Close" and is used in a dialog that already has a [X] close button